File: BrawlBuddy.Api/Models/BattleLog.cs
// BattleLog model based on Brawl Stars API structure

namespace BrawlBuddy.Api.Models;

public class BattleLog
{
    public List<BattleLogItem> Items { get; set; } = new();
}

public class BattleLogItem
{
    public string BattleTime { get; set; } = string.Empty;
    public BattleEvent Event { get; set; } = new();
    public Battle Battle { get; set; } = new();
}

public class BattleEvent
{
    public int Id { get; set; }
    public string Mode { get; set; } = string.Empty;
    public string Map { get; set; } = string.Empty;
}

public class Battle
{
    public string Mode { get; set; } = string.Empty;
    public string Type { get; set; } = string.Empty;
    public string Result { get; set; } = string.Empty;
    public int Duration { get; set; }
    public int TrophyChange { get; set; }
    public int StarTokensGained { get; set; }
    public List<List<BattlePlayer>> Teams { get; set; } = new();
}

public class BattlePlayer
{
    public string Tag { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public BattleBrawler Brawler { get; set; } = new();
}

public class BattleBrawler
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public int Power { get; set; }
    public int Trophies { get; set; }
}

File: brawlbuddy-frontend/src/types/BattleLog.ts
// BattleLog types based on backend API models
export interface BattleLog {
  items: BattleLogItem[];
}

export interface BattleLogItem {
  battleTime: string;
  event: BattleEvent;
  battle: Battle;
}

export interface BattleEvent {
  id: number;
  mode: string;
  map: string;
}

export interface Battle {
  mode: string;
  type: string;
  result: string;
  duration: number;
  trophyChange: number;
  starTokensGained: number;
  teams: BattlePlayer[][];
  players?: BattlePlayer[]; // For solo modes
  starPlayer?: BattlePlayer; // Optional star player
}

export interface BattlePlayer {
  tag: string;
  name: string;
  brawler: BattleBrawler;
}

export interface BattleBrawler {
  id: number;
  name: string;
  power: number;
  trophies: number;
}